File: frontend/src/types/payment.ts
import type { User } from './user';

export type PaymentStatus = 'unpaid' | 'paid' | 'overdue';
export type FeeType = 'property' | 'parking' | 'utilities';

export interface Payment {
  id: number;
  userId: number;
  feeType: FeeType;
  amount: number;
  month: string;
  status: PaymentStatus;
  paidAt?: string;
  createdAt: string;
  user?: User;
}

File: backend/src/main/java/com/smartestate/entity/Payment.java
package com.smartestate.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("payments")
public class Payment {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String feeType;
    private BigDecimal amount;
    private String month;
    private String status;
    private LocalDateTime paidAt;
    private LocalDateTime createdAt;

    @TableField(exist = false)
    private User user;
}
